### PR TITLE
feat: transform state to corresponding css selector and data-attr

### PR DESCRIPTION
**Pull request by sebald**

### CODE REVIEW

File: packages/system/src/hooks/useStateProps.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  [key in StateAttrsKey]?: '';
+};
+
+/**
+ * Given a map of states (e.g. `{ hover: true, focus: false }`) returns an
+ * object that can be used to set state props on a component
+ * (e.g. `[data-hover]` and `[data-focus]`).
+ */
+export const useStateProps = (
+  states: { [key in ComponentState]?: boolean } = {}
```
Comment (ti10le) [inline packages/system/src/hooks/useStateProps.ts:27]: coverage has pointed out that there is no test when {} is specified without states. So the default. Then I asked myself, should I just add the test or check in the hook if states are specified and if not show an error message? Or why should you use the hook without specifying states? What do you mean?

File: packages/system/src/components/Box/utils.test.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+          },
+          "color": "#1864ab",
+        },
+        "padding": 8,
+      }
+    `);
+  });
+});
+
+// -> add to box and test it there
```
Comment (ti10le) [inline packages/system/src/components/Box/utils.test.ts:121]: remove this comment?

**Review by ti10le — CHANGES_REQUESTED** [review]

File: packages/system/src/components/Box/utils.test.ts (hunk 2 above)
Comment (sebald) [reply packages/system/src/components/Box/utils.test.ts:121]: Yes 😅

File: packages/system/src/hooks/useStateProps.ts (hunk 1 above)
Comment (ti10le) [reply packages/system/src/hooks/useStateProps.ts:27]: But then fix coverage with a test for this, right? 😊

Comment (ti10le) [reply packages/system/src/hooks/useStateProps.ts:27]: hm.. still saying line 27 is not tested. Whats the problem?

**Review by ti10le — APPROVED** [review]

**Pull request merged**